File: SDKs/monarchinitiative.org/1.0.1/typescript/src/sdk/owlontology.ts
import { AxiosInstance, AxiosRequestConfig, AxiosResponse } from "axios";
import * as operations from "./models/operations";
import * as utils from "../internal/utils";

export class OwlOntology {
  _defaultClient: AxiosInstance;
  _securityClient: AxiosInstance;
  _serverURL: string;
  _language: string;
  _sdkVersion: string;
  _genVersion: string;

  constructor(defaultClient: AxiosInstance, securityClient: AxiosInstance, serverURL: string, language: string, sdkVersion: string, genVersion: string) {
    this._defaultClient = defaultClient;
    this._securityClient = securityClient;
    this._serverURL = serverURL;
    this._language = language;
    this._sdkVersion = sdkVersion;
    this._genVersion = genVersion;
  }
  
  /**
   * getDlQuery - Placeholder - use OWLery for now
  **/
  getDlQuery(
    req: operations.GetDlQueryRequest,
    config?: AxiosRequestConfig
  ): Promise<operations.GetDlQueryResponse> {
    if (!(req instanceof utils.SpeakeasyBase)) {
      req = new operations.GetDlQueryRequest(req);
    }
    
    const baseURL: string = this._serverURL;
    const url: string = utils.generateURL(baseURL, "/owl/ontology/dlquery/{query}", req.pathParams);
    
    const client: AxiosInstance = this._defaultClient!;
    
    const r = client.request({
      url: url,
      method: "get",
      ...config,
    });
    
    return r.then((httpRes: AxiosResponse) => {
        const contentType: string = httpRes?.headers?.["content-type"] ?? "";

        if (httpRes?.status == null) throw new Error(`status code not found in response: ${httpRes}`);
        const res: operations.GetDlQueryResponse = {statusCode: httpRes.status, contentType: contentType};
        switch (true) {
          case httpRes?.status == 200:
            if (utils.matchContentType(contentType, `application/json`)) {
                res.associations = httpRes?.data;
            }
            break;
        }

        return res;
      })
  }

  
  /**
   * getSparqlQuery - Placeholder - use direct SPARQL endpoint for now
  **/
  getSparqlQuery(
    req: operations.GetSparqlQueryRequest,
    config?: AxiosRequestConfig
  ): Promise<operations.GetSparqlQueryResponse> {
    if (!(req instanceof utils.SpeakeasyBase)) {
      req = new operations.GetSparqlQueryRequest(req);
    }
    
    const baseURL: string = this._serverURL;
    const url: string = utils.generateURL(baseURL, "/owl/ontology/sparql/{query}", req.pathParams);
    
    const client: AxiosInstance = this._defaultClient!;
    
    const r = client.request({
      url: url,
      method: "get",
      ...config,
    });
    
    return r.then((httpRes: AxiosResponse) => {
        const contentType: string = httpRes?.headers?.["content-type"] ?? "";

        if (httpRes?.status == null) throw new Error(`status code not found in response: ${httpRes}`);
        const res: operations.GetSparqlQueryResponse = {statusCode: httpRes.status, contentType: contentType};
        switch (true) {
          case httpRes?.status == 200:
            if (utils.matchContentType(contentType, `application/json`)) {
                res.associations = httpRes?.data;
            }
            break;
        }

        return res;
      })
  }

}

File: SDKs/monarchinitiative.org/1.0.1/python/src/sdk/owl_ontology.py
import requests
from typing import Optional
from sdk.models import shared, operations
from . import utils

class OwlOntology:
    _client: requests.Session
    _security_client: requests.Session
    _server_url: str
    _language: str
    _sdk_version: str
    _gen_version: str

    def __init__(self, client: requests.Session, security_client: requests.Session, server_url: str, language: str, sdk_version: str, gen_version: str) -> None:
        self._client = client
        self._security_client = security_client
        self._server_url = server_url
        self._language = language
        self._sdk_version = sdk_version
        self._gen_version = gen_version

    
    def get_dl_query(self, request: operations.GetDlQueryRequest) -> operations.GetDlQueryResponse:
        r"""Placeholder - use OWLery for now
        """
        
        base_url = self._server_url
        
        url = utils.generate_url(base_url, "/owl/ontology/dlquery/{query}", request.path_params)
        
        
        client = self._client
        
        r = client.request("GET", url)
        content_type = r.headers.get("Content-Type")

        res = operations.GetDlQueryResponse(status_code=r.status_code, content_type=content_type)
        
        if r.status_code == 200:
            if utils.match_content_type(content_type, "application/json"):
                out = utils.unmarshal_json(r.text, Optional[list[shared.Association]])
                res.associations = out

        return res

    
    def get_sparql_query(self, request: operations.GetSparqlQueryRequest) -> operations.GetSparqlQueryResponse:
        r"""Placeholder - use direct SPARQL endpoint for now
        """
        
        base_url = self._server_url
        
        url = utils.generate_url(base_url, "/owl/ontology/sparql/{query}", request.path_params)
        
        
        client = self._client
        
        r = client.request("GET", url)
        content_type = r.headers.get("Content-Type")

        res = operations.GetSparqlQueryResponse(status_code=r.status_code, content_type=content_type)
        
        if r.status_code == 200:
            if utils.match_content_type(content_type, "application/json"):
                out = utils.unmarshal_json(r.text, Optional[list[shared.Association]])
                res.associations = out

        return res

    